test/common/testlib.py

```diff
@@ -1545,7 +1559,11 @@ def test_main(options=None, suite=None, attachments=None, **kwargs):
     If no arguments are given on the command line, all test cases are
     executed.  Otherwise only the given test cases are run.
     """
-
+    # find who called me, to be able to skip this as a test in case of
+    # pytest as scheduler
+    who_runs_me = inspect.getmodule(inspect.stack()[1][0]).__name__
+    if "pytest" in who_runs_me:
+        raise unittest.SkipTest("skip test_main when using pytest as test scheduler")
```

Comment:
I don't understand -- this just skips every test with pytest?

Reply:
becase `test*` is basic pattern for tests, so that when there is import of test_main in each `check` it is imported as test as well. and this cause that when invoked via pytest, this `test_main` is ignored

Reply:
Ah, I understand -- it doesn't check whether something is a subclass of `unittest.TestCase`, it just goes by name?

Reply:
yes, it found name of second item of caller stack and try to find it there is this substring as a name. Original run-test or direct call has there `__main__` as an name. calling via pytest has there `_pytest.*` name